lib/dor/datastreams/content_metadata_ds.rb

```diff
@@ -35,10 +35,40 @@ class ContentMetadataDS < ActiveFedora::OmDatastream
 
     def public_xml
       result = ng_xml.clone
-      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")])]'   ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'      ).each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver').each { |n| n.remove }
-      result.xpath('/contentMetadata/resource/file/checksum').each { |n| n.remove }
+      
+      # remove any resources or attributes that are not destined for the public XML
+      result.xpath('/contentMetadata/resource[not(file[(@deliver="yes" or @publish="yes")]|externalFile)]').each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file[not(@deliver="yes" or @publish="yes")]'                ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file').xpath('@preserve|@shelve|@publish|@deliver'          ).each { |n| n.remove }
+      result.xpath('/contentMetadata/resource/file/checksum'                                              ).each { |n| n.remove }
```

Comment:
If it's not too much trouble, add the additional specs for 41-43 in this PR.  Otherwise, well, please create the tech-debt issue and refer to this PR there and the coverage concern noted here.